src/averageAges.js

```diff
@@ -55,7 +72,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => onlyWithSon ? people
+    .some(data => person.mother === data.name && person.sex === 'm')
```

Comment:
```suggestion
    .some(data => person.mother === data.name) && person.sex === 'm'
```
person.sex should be checked outside of `some` method